Fred2/IO/MartsAdapter.py

```diff
@@ -101,8 +101,8 @@ def get_product_sequence(self, product_id, **kwargs):
             + self.biomart_tail
 
         # logging.warn(rq_n)
-        tsvreader = csv.DictReader(urllib.request.urlopen(self.biomart_url
-                                                   + urllib.parse.quote(rq_n)).read().splitlines(), dialect='excel-tab')
+        tsvreader = csv.DictReader((urllib.request.urlopen(self.biomart_url +
```

Comment:
_csv.Error: iterator should return strings, not bytes (did you open the file in text mode?) 
In Python 2, the csv module requires you to open files in binary mode. This is because the module needs to be able to control line-endings, as typical CSV files use DOS line-endings, and the text-mode under Python 2 can change line-endings on some platforms. The csv module will also return and expect data in byte-strings.
The Python 3 csv-module instead requires you to open the file in text-mode with newline='', and it returns and expects Unicode strings. [http://python3porting.com/problems.html]
https://stackoverflow.com/questions/18897029/read-csv-file-from-url-into-python-3-x-csv-error-iterator-should-return-str https://stackoverflow.com/questions/16283799/how-to-read-a-csv-file-from-a-url-with-python/62614979#62614979
https://stackoverflow.com/questions/6862770/let-json-object-accept-bytes-or-let-urlopen-output-strings